test/e2e/networking_perf.go

```diff
@@ -0,0 +1,249 @@
+				Image: "mattgruter/iperf",
+				Args: []string{
+					"-s", "-p", "8001"},
+				Ports: []api.ContainerPort{{ContainerPort: 8001}},
+			},
+			// this will be used to generate the -service name which all iperf clients point at.
+			numServer, // 1 server
+		)
+		CreateServiceWithPods(f,
+			-1,
+			-1,
+			"iperf-e2e-cli", // this will be used to generate the -service name which all iperf clients point at.
+			api.Container{
+				Name:  "iperf-client",
+				Image: "mattgruter/iperf",
+				Args: []string{
+					"-c", "service-for-" + appName, "-p", "8002", "--reportstyle", "C",
+				},
+			},
+			numClient, // 1 server
+		)
+		Logf("Reading all perf results to stdout.")
+		Logf("date,cli,cliPort,server,serverPort,id,interval,transferBits,bandwidthBits")
+		bandwidthMap := make(map[string]int64)
+
+		// Calculate expected number of clients based on total nodes.  ITs own function since this
+		// logic may change over time at scale.
+		expectedCli := func() int {
+			nodes, err := GetReadyNodes(f)
+			expectNoError(err)
+			return int(math.Min(float64(len(nodes.Items)), float64(numClient)))
+		}()
+
+		start := time.Now()
+		for len(bandwidthMap) < expectedCli && time.Since(start) <= 2*time.Minute {
+			TryEachPod(f.Client, f.Namespace.Name, "app", "iperf-e2e-cli-pod",
+				func(p api.Pod) {
+					resultS, err := lookForStringInLog(f.Namespace.Name, p.Name, "iperf-client", "0-10.0", 1*time.Second)
+					if err == nil {
+						Logf(resultS)
+						ipr := NewIPerf(resultS)
+						bandwidthMap[p.Name+"-"+ipr.cli+"-"+ipr.id+"--server="+ipr.server] = ipr.bandwidthBits
+						if len(bandwidthMap) >= expectedCli {
+							Logf("Got enough results (%v / %v) to terminate the test...", len(bandwidthMap), expectedCli)
+						} else {
+							Logf("Only got results from %v out of %v...", len(bandwidthMap), expectedCli)
+						}
+					} else {
+						Logf("No result for %v. ", p.Name)
+					}
+				})
+			time.Sleep(1 * time.Second)
+		}
+
+		fmt.Println("[begin] Node,Bandwith CSV")
+		for node, bandwidth := range bandwidthMap {
+			fmt.Println(node, "\t ", fmt.Sprintf("%E", float64(bandwidth)))
+		}
+		fmt.Println("[end] Node,Bandwith CSV")
+
+		for node, bandwidth := range bandwidthMap {
+			Logf("%v had bandwidth %v.  Ratio to expected (%v) was %f", node, bandwidth, expectedBandwidth, float64(bandwidth)/float64(expectedBandwidth))
+		}
+	})
+}
+
+// A struct modelling an iperf record.  The records look something like this.
+// The struct fields are in order !
+// 20160314154239,172.17.0.3,34152,172.17.0.2,5001,3,0.0-10.0,33843707904,27074774092
+type IPerfResult struct {
+	date          string // field 1 in the csv
+	cli           string // field 2 in the csv
+	cliPort       int64  // ...
+	server        string
+	servPort      int64
+	id            string
+	interval      string
+	transferBits  int64
+	bandwidthBits int64
+}
+
+// parse parses an IPerf CSV output line into an IPerfResult
+func NewIPerf(csvLine string) IPerfResult {
```

Comment:
This seems rigidly tied to the version of iperf.  

Is there a more verbose output that could be more resilient to changes in iperf? e.g. name=value